### PR TITLE
validate: add args validation

### PR DESCRIPTION
**Pull request by Mashimiao**
Signed-off-by: Ma Shimiao <mashimiao.fnst@cn.fujitsu.com>

### CODE REVIEW

**Comment by liangchenye** [comment]
LGTM

File: validate/validate.go [hunk 1]
```diff
@@ -218,6 +218,23 @@ func (v *Validator) CheckProcess() (msgs []string) {
...
+		if filepath.IsAbs(process.Args[0]) {
+			var rootfsPath string
+			if filepath.IsAbs(v.spec.Root.Path) {
+				rootfsPath = v.spec.Root.Path
+			} else {
+				rootfsPath = filepath.Join(v.bundlePath, v.spec.Root.Path)
+			}
+			absPath := filepath.Join(rootfsPath, process.Args[0])
+			if _, err := os.Stat(absPath); err != nil {
+				msgs = append(msgs, err.Error())
```
Comment (wking) [inline validate/validate.go]: Not finding the binary in the `root.path` is not necessarily fatal.  Perhaps a later mount or pre-start hook or some such was going to provide a binary at the configured path (previous discussion [here][1]).

[1]: https://github.com/opencontainers/runc/pull/827#r63464201

**Comment by Mashimiao** [comment]
@wking  @liangchenye  @mrunalp  @hqhq  PTAL

File: validate/validate.go [hunk 2]
```diff
@@ -218,6 +218,33 @@ func (v *Validator) CheckProcess() (msgs []string) {
...
+			var rootfsPath string
+			if filepath.IsAbs(v.spec.Root.Path) {
+				rootfsPath = v.spec.Root.Path
+			} else {
+				rootfsPath = filepath.Join(v.bundlePath, v.spec.Root.Path)
+			}
+			absPath := filepath.Join(rootfsPath, process.Args[0])
+			fileinfo, err := os.Stat(absPath)
+			if err != nil {
+				if !os.IsNotExist(err) {
```
Comment (wking) [inline validate/validate.go]: nit: You can flatten this a bit with:

```go
if os.IsNotExist(err) {
  …
} else if err != nil {
  …
} else {
  …
}
```

Comment (Mashimiao) [inline validate/validate.go]: Fine.

**Comment by Mashimiao** [comment]
@wking  @liangchenye @mrunalp @hqhq PTAL

**Comment by wking** [comment]
113fd04 looks fine to me.

**Comment by mrunalp** [comment]
LGTM

**Comment by hqhq** [comment]
LGTM

**Pull request merged**